derive_builder_core/src/setter.rs

```diff
@@ -146,14 +154,19 @@ impl<'a> ToTokens for Setter<'a> {
                     quote!(<VALUE: ::derive_builder::export::core::convert::TryInto<#ty>>);
                 let try_ident = syn::Ident::new(&format!("try_{}", ident), Span::call_site());
 
+                let mut converted = quote! {converted};
+                if builder_field_is_option {
+                    converted = wrap_expression_in_some(converted);
+                }
```

Comment:
Can this be rewritten as `let converted = if builder_field_is_option { ... } else { ... }`?

Reply:
Well, it could be
```
    let converted = if builder_field_is_option {
        wrap_expression_in_some(quote! {converted})
    } else {
        quote! {converted}
    };
```
but that has `quote! {converted}` twice.  Or
```
    let converted = quote! {converted};
    let converted = builder_field_is_option {
        wrap_expression_in_some(converted)
    } else {
        converted
    };
```
which seems hardly better, if at all.

In fact *every* call to `wrap_expression_in_some` is conditional and uses the pattern with `mut`: here, and (earlier) the two successive calls on `into_value`.

How about
```
    if (stripped_option) { wrap_expression_in_some(&mut into_value); }
...
    if (builder_field_is_option) { wrap_expression_in_some(&mut converted); }
```

Reply:
Hmm; I don't mind the `quote!(converted)` appearing twice, but I don't feel strongly enough about it to delay this PR.